homeassistant/components/sensor/buienradar.py

```diff
@@ -0,0 +1,471 @@
+CONF_BR_ZIN = '@zin'
+CONF_BR_IMG = '_img'
+CONF_BR_FORECAST = 'verwachting_meerdaags'
+CONF_BR_DAYFC = "dag-plus%d"
+CONF_BR_MINTEMP = 'maxtemp'
+CONF_BR_MAXTEMP = 'maxtempmax'
+
+# Sensor types are defined like so:
+# SENSOR_TYPES = { 'key': ['Display name',
+#                          'unit of measurement',
+#                          'key in buienradar xml'],}
+SENSOR_TYPES = {
+    CONF_STATIONNAME: ['Stationname', None, 'stationnaam'],
+    CONF_SYMBOL: ['Symbol', None, 'icoonactueel'],
+    CONF_HUMIDITY: ['Humidity', '%', 'luchtvochtigheid'],
+    ATTR_TEMPERATURE: ['Temperature', '°C', 'temperatuurGC'],
+    CONF_GROUNDTEMP: ['Ground Temperature', '°C', 'temperatuur10cm'],
+    CONF_WINDSPEED: ['Wind speed', 'm/s', 'windsnelheidMS'],
+    CONF_WINDFORCE: ['Wind force', 'Bft', 'windsnelheidBF'],
+    CONF_WINDDIRECTION: ['Wind direction', '°', 'windrichtingGR'],
+    CONF_WINDAZIMUTH: ['Wind direction azimuth', None, 'windrichting'],
+    CONF_PRESSURE: ['Pressure', 'hPa', 'luchtdruk'],
+    CONF_VISIBILITY: ['Visibility', 'm', 'zichtmeters'],
+    CONF_WINDGUST: ['Wind gust', 'm/s', 'windstotenMS'],
+    CONF_PRECIPITATION: ['Precipitation', 'mm/h', 'regenMMPU'],
+    CONF_IRRADIANCE: ['Irradiance', 'W/m2', 'zonintensiteitWM2'],
+}
+
+SENSOR_ICONS = {
+    CONF_HUMIDITY: 'mdi:water-percent',
+    ATTR_TEMPERATURE: 'mdi:thermometer',
+    CONF_GROUNDTEMP: 'mdi:thermometer',
+    CONF_WINDSPEED: 'mdi:weather-windy',
+    CONF_WINDFORCE: 'mdi:weather-windy',
+    CONF_WINDDIRECTION: 'mdi:compass-outline',
+    CONF_WINDAZIMUTH: 'mdi:compass-outline',
+    CONF_PRESSURE: 'mdi:gauge',
+    CONF_WINDGUST: 'mdi:weather-windy',
+    CONF_IRRADIANCE: 'mdi:sunglasses',
+    CONF_PRECIPITATION: 'mdi:weather-pouring',
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_MONITORED_CONDITIONS,
+                 default=[CONF_SYMBOL, ATTR_TEMPERATURE]): vol.All(
+                     cv.ensure_list, vol.Length(min=1),
+                     [vol.In(SENSOR_TYPES.keys())]),
+    vol.Optional(CONF_LATITUDE): cv.latitude,
+    vol.Optional(CONF_LONGITUDE): cv.longitude,
+})
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Setup the buienradar_nl sensor."""
+    latitude = config.get(CONF_LATITUDE, hass.config.latitude)
+    longitude = config.get(CONF_LONGITUDE, hass.config.longitude)
+
+    if None in (latitude, longitude):
+        _LOGGER.error("Latitude or longitude not set in HomeAssistant config")
+        return False
+
+    coordinates = {CONF_LATITUDE: float(latitude),
+                   CONF_LONGITUDE: float(longitude)}
+
+    dev = []
+    for sensor_type in config[CONF_MONITORED_CONDITIONS]:
+        dev.append(BrSensor(sensor_type))
+    async_add_devices(dev)
+
+    weather = BrData(hass, coordinates, dev)
+    # Update weather every 10 minutes, since
+    # the data gets updated every 10 minutes
+    async_track_time_interval(hass, weather.async_update,
```

Comment:
yes; i see your point....
This seems to be a leftover from the platform I used as a basis for this one.....
I will optimize this in the next version.

Is the preferred pattern to use something like this:
```
nxt = dt_util.utcnow() + timedelta(minutes=X)
async_track_point_in_utc_time(self.hass, this.async_update, nxt)
```

where the value of next is determined by success (10 min) of failure (2 minutes)?